tooling-mp-jwt-auth-tool/src/main/java/org/jboss/eap/qe/microprofile/jwt/auth/tool/AbstractKeyLoader.java

```diff
@@ -0,0 +1,73 @@
+package org.jboss.eap.qe.microprofile.jwt.auth.tool;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.security.KeyFactory;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+
+import org.jose4j.base64url.internal.apache.commons.codec.binary.Base64;
+
+/**
+ * Utility class for reading private and public keys from pem files;
+ * Note the keys have to be in `pkcs8` format (Public Key Cryptography Standards #8 (PKCS#8) PEM format)
+ * which is thr format supported by MP JWT (see "Supported Public Key Formats" in "Eclipse MicroProfile Interoperable
+ * JWT RBAC");
+ *
+ * @param <PRIVATE_KEY> : Private Key interface for the algorithm in use
+ * @param <PUBLIC_KEY> : Public Key interface for the algorithm in use
+ * @author tborgato
+ */
+public abstract class AbstractKeyLoader<PRIVATE_KEY, PUBLIC_KEY> {
```

Comment:
Isn't this redundant to `org.jboss.eap.qe.microprofile.jwt.auth.tool.RsaKeyTool`?

Reply:
probably yes, but I needed it in order to handle the new EC algorithm

Reply:
I mean I had to add code for the new EC algorithm somewhere and, since we'll probably progressively introduce `jose4j`, I preferred to keep it separated in order ease pruning in some future .... I can try and merge the two but it would take some time since I'd have to revise and test all places where it's used .... I can do it but if you agree I'll do it in some subsequent MR since this RFE needs to be ready by end of next week

Reply:
From my point of view is we now have bunch of redundant code which does exactly the same thing. I'm really not fan of subsequent MRs because then it will be another record in issues and when merged, this redundancy will bring the code maintainability down.

Reply:
@honza-kasik doing the complete migration in this MR would take a lot of time .... I can deprecated things in `RsaKeyTool` to make the intent evident ...

Reply:
@tommaso-borgato Yes please, deprecate it then and properly document it.

Reply:
@honza-kasik done

Reply:
Thanks! LGTM